vertx-web/src/main/java/io/vertx/ext/web/impl/RouteImpl.java

```diff
@@ -295,35 +297,19 @@ synchronized boolean matches(RoutingContext context, String mountPoint, boolean
     }
     if (!consumes.isEmpty()) {
       // Can this route consume the specified content type
-      String contentType = request.headers().get("content-type");
-      boolean matches = false;
-      for (String ct: consumes) {
-        if (ctMatches(contentType, ct)) {
-          matches = true;
-          break;
-        }
-      }
-      if (!matches) {
-        return false;
-      }
+      // MYTODO How should this be? Cast or add to interface?
+      ParsedMIME contentType = ((HttpServerRequestWrapper) request).parsedHeaders().contentType;
```

Comment:
Alternatively we could add this functionality to the RoutingContext since its where all the non-basic functionality is added e.g. language support